### PR TITLE
Possibly fixing navigation issue

### LINKED ISSUE #712: Page/layout navigation broken in rare circumstances
### Description of the bug

After merging #698, @StianVestli found a very strange bug where navigating to the next page would trigger navigation to wildly different pages. I received a redux action log showing the problem, and it seems `updateCurrentView` is triggered, requesting navigation to page 2, but then it ends by triggering a `updateCurrentViewFulfilled` navigating to page 7 instead. The `newView` variable is cleanly passed through `updateCurrentViewSaga`, so I struggle to figure out exactly why this happened, or how to reproduce the problem.

### Steps To Reproduce

According to what we seem to know at this point (beware, I'm still unable to reliably reproduce this):

You seem to need an app configured with:
- Tracks, using the server-side PageOrder functionality
- Navigation buttons triggering page order calculation
- At least one page that is hidden/visible using expressions

1. Do not clear your cookies!
2. Open an app, fill out all pages, submit the instance.
3. Restart the app with a new instance (how exactly is not clear to me yet, there are multiple ways)
4. Start filling out the form again, observe that navigating to the next page may send you to entirely different pages

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by olemartinorg**
## Description
Experimental fix for the referenced navigation issue. Unsure if this fixes the problem, but it should/could be an improvement.

What this does, in a nutshell:
1. Changing the way we wait before fulfilling navigation to the next page until data is saved. Using `waitFor` instead of saga logic. Since the saga has two `take` calls, I suspected it might take an entirely different (previously triggered?) action in the queue.
2. If we forcibly navigate to another page because the current one is hidden (see #698), skip all logic and just navigate to that page instantly.
3. Useless unit tests for the now removed `watchUpdateCurrentViewSaga` has been removed 
4. Nice-to-have: Scroll to the top when navigation has been fulfilled.

## Related Issue(s)
- closes #712

## Verification
- Manual testing
  - [x] I have tested these changes manually (or: I have tested that my changes still allows navigation between pages)
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added
  - [ ] Cypress E2E test(s) have been added
  - [x] No automatic tests are needed here (or: I can't create automated regression tests for an issue I'm unable to even manually reproduce)
  - [ ] I want someone to help me make some tests
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been updated
   
  - [x] No changes/updates needed
- Changes/additions to component properties
  - [ ] Changes are reflected in both `layout/index.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [x] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
   
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board

### CODE REVIEW

File: src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts [hunk 1]
```diff
@@ -529,19 +533,6 @@ export function* watchInitialCalculatePageOrderAndMoveToNextPageSaga(): SagaIter
   }
 }
 
-export function* watchUpdateCurrentViewSaga(): SagaIterator {
```
Comment (Magnusrm) [inline src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts:532]: Husker denne ble lagt til for at backend-valideringer ikke skulle kjøre på utdatert skjemadata. Det var noe greier med at hvis viewet ble oppdatert før ulagrede data var lagret, så ville ikke valideringsmeldingen fra backend dukke opp ettesom den valideringen ikke hadde fått kjørt enda. Vet ikke hvordan dette funker nå, men kan være verdt å sjekke ut.
https://github.com/Altinn/app-frontend-react/pull/187

**Review by Magnusrm — APPROVED** [review]
Looks good! 🥇 View comment

File: src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts (hunk 1 above)
Comment (olemartinorg) [reply src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts:532]: Oooh! Spenstig - den delen av historien fikk ikke jeg med meg. Interessant! Kan forstå at det var vanskelig å lage en cypress-test for å verifisere at feilen var fikset den gang - men det hadde vært gunstig nå. De unit-testene som lå inne testet jo egentlig at _oppførselen til funksjonen var slik som den var implementert_, og det testet ikke at _oppførselen til app-frontend nå var som forventet_.

Jeg prøvde å beholde oppførselen (ved å legge inn `waitFor`) slik at man fortsatt ikke navigerer til neste side før data er lagret, men dette bør testes skikkelig tenker jeg. Jeg prøver å lage en relevant cypress-test for dette.

**Review by Magnusrm — APPROVED** [review]
LGTM 👑

**Pull request merged**